fuel-tests/tests/tx.rs

```diff
@@ -121,6 +121,29 @@ async fn submit() {
     assert_eq!(tx.id(), ret_tx.id());
 }
 
+#[tokio::test]
+async fn submit_utxo_verified_tx() {
+    let config = Config {
+        utxo_validation: true,
+        ..Config::local_node()
+    };
+
+    let srv = FuelService::new_node(config).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    let tx = Transaction::default();
+
+    let id = client.submit(&tx).await.unwrap();
+    // verify that the tx returned from the api matches the submitted tx
+    let ret_tx = client
+        .transaction(&id.0.to_string())
+        .await
+        .unwrap()
+        .unwrap()
+        .transaction;
+    assert_eq!(tx.id(), ret_tx.id());
```

Comment:
Should the test verify that the transaction was actually included in a block? Just returning the transaction object submitted via the API isn't a guarantee that the node is operating properly with the new txpool.

Reply:
`submit_tx` also checks this way, however I definitely agree the block is a better source of truth than just a returned transaction, will modify my test to check the block

Reply:
Test should check to ensure the transaction was actually included in the block now

Reply:
Transparent transaction is just a different client encoding type (json instead of binary). We should actually check the `TransactionStatus` variant if it's successful. `client.transaction_status`

Reply:
TransactionStatus looks like this:
```
pub enum TransactionStatus {
    Submitted {
        submitted_at: DateTime<Utc>,
    },
    Success {
        block_id: String,
        time: DateTime<Utc>,
        program_state: ProgramState,
    },
    Failure {
        block_id: String,
        time: DateTime<Utc>,
        reason: String,
        program_state: Option<ProgramState>,
    },
}
```

So we can confirm it was included in a block if we get the `Success` variant and the block_id is real.

Reply:
We should also test that the pool is still well behaved after submitting multiple transactions. If we don't verify all these things properly then it could break the SDK if we release this.

Reply:
> We should also test that the pool is still well behaved after submitting multiple transactions, since we had to add logic in this PR to manually remove txs from the pool after including them in a block. If we don't verify all these things properly then it could break the SDK if we release this.

If there is bug in txpool, shound't we just fix it if problem arises? Usage by sdk will be good test.